aptos-move/move-examples/dao/nft_dao/sources/nft_dao.move

```diff
@@ -0,0 +1,639 @@
+        resolve_threshold: ResolveThreshold,
+        /// The NFT Collection that is used to govern the DAO
+        governance_token: GovernanceToken, // This is the governance token or NFT used for voting
+        /// The address of the creator account this DAO owns. THis account is a resource account that this DAO is an adminstrator of the controlled account
+        controlled_account: address, // This is the nft creator account that this DAO governs, this accout should offer it signer cap to DAO account
+        /// The voting duration in secs
+        voting_duration: u64,
+        /// Total number of eligible member NFTs
+        total_supply: u64,
+        /// Minimum weight for proposal
+        min_proposal_weight: u64,
+        /// Proposal counter
+        cur_largest_proposal_id: u64,
+    }
+
+    struct ResolveThreshold has store {
+        numerator: u64,
+        denominator: u64,
+    }
+
+    /// The collection should only contains NFTs, where all token name only has 1 maximal and immutable
+    /// The total supply is fixed with the token names.
+    struct GovernanceToken has store {
+        /// The creator address of this NFT collection
+        creator: address,
+        /// The collection name
+        collection: String,
+        /// Eligible token_ids that can vote. Value of the table is the weights of the token
+        member_tokens: Table<TokenId, u64>
+    }
+
+    /// All proposals
+    struct Proposals has key {
+        proposals: Table<u64, Proposal>,
+    }
+
+    /// Store the general information about a proposal
+    struct Proposal has copy, drop, store {
+        /// Name of the proposal, limiting to 64 chars
+        name: String,
+        /// Description of the proposal, limiting to 512 chars
+        description: String,
+        /// The name of function to be executed
+        function_name: String,
+        /// The function arguments to be exectued
+        function_args: PropertyMap,
+        /// The start time of the voting
+        start_time_sec: u64,
+        /// Proposal results, unresolved, passed, not passed
+        resolution: u8,
+    }
+
+    struct ProposalVotingStatistics has key {
+        proposals: Table<u64, VotingStatistics>,
+    }
+
+    struct VotingStatistics has store {
+        /// Total yes votes
+        total_yes: u64,
+        /// Total no notes
+        total_no: u64,
+        /// Token voted yes
+        yes_votes: BucketTable<TokenId, address>, // address is the original voter's address for keeping a record of who voted
+        /// Token voted no
+        no_votes: BucketTable<TokenId, address>,
+    }
+
+    struct RegisteredAccounts has key {
+        /// Key is the address of DAO creator's account and value is the signer capability of its resource account
+        /// Only one DAO per admin address for now.
+        accounts_signer_caps: Table<address, SignerCapability>,
+        res_acct_to_src_acct: Table<address, address>,
```

Comment:
What is this for?

Reply:
reverse look-up in the resolve function to find the source account of a resource account, which is then used to locate the signer-capability.

Reply:
This can be avoidable per reply in another comment